src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Scanner;
+
+/**
+ * A class that creates the Storage object.
+ */
+public class Storage {
+    private String pathName;
+    private File file;
+    private TaskList previousTaskList;
+
+    /**
+     * A constructor that intialises the Storage object.
+     *
+     * @param pathName Path name of the text file where the list of tasks is stored.
+     * @param previousTaskList ArrayList of tasks.
+     * @throws IOException If an I/O error occurred.
+     */
+    public Storage(String pathName, TaskList previousTaskList) throws IOException {
+        this.pathName = pathName;
+        this.file = new File(pathName);
+        this.previousTaskList = previousTaskList;
+
+        if (!Files.exists(Path.of(pathName))) {
+            file.createNewFile();
```

Comment:
I like that you handled the case if the file does not already exists. However is the pathName OS-independent as required by the project outline? (Could use Paths.get() to fetch the relative file path then append the given filePath at the back to form the full file path for storage)